### PR TITLE
Registration form csv

### PR DESCRIPTION
**Pull request by justinlittman**
Rebase after PR #3653 is merged;  This has already been vetted by Andrew as part of vetting PR #3653, so it is ready to be merged once #3653 is merged.

## Why was this change made? 🤔
CSV are easier than forms.

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact (including writing to shared file systems or interacting with other SDR APIs), ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

⚡ ⚠ If this change updates the Argo UI, run all integration tests that use Argo and create a PR on the integration test repo to fix anything this change breaks. ⚡

### CODE REVIEW

**Comment by ndushay** [comment]
Will this ix #3611?

**Comment by justinlittman** [comment]
The magic keyword is in the git commit for this PR.

**Comment by ndushay** [comment]
> The magic keyword is in the git commit for this PR.

does that connect the PR to the issue on github and/or close the issue automagically?  If so ... wow!

File: app/forms/csv_registration_form.rb [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  class VirtualModel < Hash
+    def persisted?
+      false
+    end
+  end
+
+  property :current_user, virtual: true
+  property :admin_policy, virtual: true
+  property :collection, virtual: true
+  property :rights, virtual: true
```
Comment (jcoyne) [inline app/forms/csv_registration_form.rb:14]: This should be view/download/location/cdl.

File: spec/features/item_registration_csv_spec.rb [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      allow(BulkAction).to receive(:new).and_return(bulk_action)
+    end
+
+    it 'starts bulk action' do
+      visit registration_path
+      select '[Internal System Objects]', from: 'Admin Policy' # "uber APO"
+      select 'registrationWF', from: 'Initial Workflow'
+      select 'book', from: 'Content Type'
+      select 'left-to-right', from: 'Viewing Direction'
+      select 'Stanford', from: 'Rights'
```
Comment (jcoyne) [inline spec/features/item_registration_csv_spec.rb:25]: The rights field no longer exists

File: spec/features/item_registration_csv_spec.rb [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  end
+
+  context 'registration fails validation' do
+    it 'reports error' do
+      visit registration_path
+      select '[Internal System Objects]', from: 'Admin Policy' # "uber APO"
+      select 'registrationWF', from: 'Initial Workflow'
+      select 'book', from: 'Content Type'
+      select 'left-to-right', from: 'Viewing Direction'
+      select 'Stanford', from: 'Rights'
```
Comment (jcoyne) [inline spec/features/item_registration_csv_spec.rb:69]: The Rights field no longer exists.

**Review by jcoyne — CHANGES_REQUESTED** [review]

**Comment by ndushay** [comment]
@justinlittman gentle reminder that this PR is still open - I'm assuming it needs PO review, so I've assigned that label but I'm not sure if there is more to do here or not (it's a draft)

**Comment by justinlittman** [comment]
This is part of the registration form redesign (#3653). @jcoyne how is this moving forward?

**Comment by ndushay** [comment]
@jcoyne is this PR the combination of JLitt's work, and of your work in PR #3653?

**Comment by ndushay** [comment]
Andrew reviewed this while reviewing PR #3653, so this has PO approval.

**Comment by ndushay** [comment]
approved by Andrew.

**Pull request merged**